src/stream_base.cc

```diff
@@ -679,9 +679,14 @@ void EmitToJSStreamListener::OnStreamRead(ssize_t nread, const uv_buf_t& buf_) {
   }
 
   CHECK_LE(static_cast<size_t>(nread), bs->ByteLength());
-  bs = BackingStore::Reallocate(isolate, std::move(bs), nread);
-
-  stream->CallJSOnreadMethod(nread, ArrayBuffer::New(isolate, std::move(bs)));
+  std::unique_ptr<BackingStore> new_bs =
```

Comment:
This can be guarded with a `static_cast<size_t>(nread) != bs->ByteLength()` too.